homeassistant/components/lg_thinq/__init__.py

```diff
@@ -3,22 +3,39 @@
 from __future__ import annotations
 
 import asyncio
+from dataclasses import dataclass, field
 import logging
 
 from thinqconnect import ThinQApi, ThinQAPIException
 from thinqconnect.integration import async_get_ha_bridge_list
 
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_ACCESS_TOKEN, CONF_COUNTRY, Platform
+from homeassistant.const import (
+    CONF_ACCESS_TOKEN,
+    CONF_COUNTRY,
+    EVENT_HOMEASSISTANT_STOP,
+    Platform,
+)
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.exceptions import ConfigEntryNotReady
 from homeassistant.helpers import device_registry as dr
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.event import async_track_time_interval
 
-from .const import CONF_CONNECT_CLIENT_ID
+from .const import CONF_CONNECT_CLIENT_ID, MQTT_SUBSCRIPTION_INTERVAL
 from .coordinator import DeviceDataUpdateCoordinator, async_setup_device_coordinator
+from .mqtt import ThinQMQTT
 
-type ThinqConfigEntry = ConfigEntry[dict[str, DeviceDataUpdateCoordinator]]
+
+@dataclass(kw_only=True)
+class ThinqData:
+    """A class that holds runtime data."""
+
+    coordinators: dict[str, DeviceDataUpdateCoordinator] = field(default_factory=dict)
+    mqtt_client: ThinQMQTT | None = None
```

Comment:
joostlek reviewed>
The class is always created right? Should we make it not nullable?

My comment>
What if mqtt is not working, but the other cloud functions are working well temporarily?
Do we have to stop other functions?
If we want to keep the functions to work, we should allow None for 'mqtt_client'.